internal/pkg/agent/application/actions/handlers/handler_action_upgrade.go

```diff
@@ -26,15 +28,17 @@ type Upgrade struct {
 	bkgCancel  context.CancelFunc
 	bkgMutex   sync.Mutex
 
-	tamperProtectionFn func() bool // allows to inject the flag for tests, defaults to features.TamperProtection
+	tamperProtectionFn         func() bool                                                                                                                            // allows to inject the flag for tests, defaults to features.TamperProtection
+	notifyUnitsOfProxiedAction func(ctx context.Context, log *logp.Logger, action dispatchableAction, ucs []unitWithComponent, performAction performActionFunc) error // allows to inject the function for tests, defaults to notifyUnitsOfProxiedAction
```

Comment:
```suggestion
	notifyUnitsOfProxiedActionFn func(ctx context.Context, log *logp.Logger, action dispatchableAction, ucs []unitWithComponent, performAction performActionFunc) error // allows to inject the function for tests, defaults to notifyUnitsOfProxiedAction
```

The type and the actual implementation having the same name confused me briefly until I realized that's what was happening, staring the assignment below I could figure out where the default value came from until I checked out the code.

Reply:
changed here 71181e28cff3e12d9bf1f3b01c2f9ec39f44b52a